Fixes cargo barcode payouts paying 50x what the sale is worth

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Makes it so sales barcodes set the modifier the pricetag component uses to calculate the sale's cut correctly again.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

People will no longer receive 50x the value of items sold with barcodes on the Supply shuttle.

Fixes #56842
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: cacogen
fix: You can no longer receive 50x the value of items sold with barcodes on the Supply shuttle
tweak: Sales barcode profit splits will consistently default to 50% (you can still set this to less, but there's no reason to at the moment)
tweak: Sales tagger examine message improvements
tweak: Sales tagger and GrandArk Exporter tablet app both tell you the valid range to set the profit split in now
spellcheck: 'Barcode tag' is now called 'barcode tag'
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
Fixes cargo barcode payouts

Mothblocks commented on code/modules/recycling/sortingmachinery.dm:
```diff
@@ -458,9 +458,9 @@
-	to_chat(user, "<span class='notice'>[percent_cut]% profit will be received if a package with a barcode is sold.</span>")
+		cut_multiplier = 0.2
+	cut_multiplier = clamp(round(potential_cut, 1), 1, 50) / 100
+	to_chat(user, "<span class='notice'>[cut_multiplier*100]% profit will be received if a package with a barcode is sold.</span>")
```
I foresee weird things with floats should the number ever be something not pretty (10.0000000000000000000001% profit). I recommend rounding this number before you display it.

cacogen replied:
```suggestion
	to_chat(user, "<span class='notice'>[round(cut_multiplier*100)]% profit will be received if a package with a barcode is sold.</span>")
```

LemonInTheDark commented on code/modules/recycling/sortingmachinery.dm:
```diff
@@ -458,9 +458,9 @@
-		percent_cut = 50
-	percent_cut = clamp(round(potential_cut, 1), 1, 50)
-	to_chat(user, "<span class='notice'>[percent_cut]% profit will be received if a package with a barcode is sold.</span>")
+		cut_multiplier = 0.2
```
doesn't this change behavior? or was it unintentionally set to 50%

cacogen replied:
I made it more consistent. You can still set this to anything between 1% and 50%. In some cases barcode values defaulted to 50% and in others 20%. I thought 20% made more sense to start off at.

cacogen replied:
Actually I've changed it to default to 50% "split" (has never actually affected cargo) and use initial() to set the var if the player doesn't input anything in these cases

LemonInTheDark commented on code/modules/recycling/sortingmachinery.dm:
```diff
@@ -470,4 +470,4 @@
 	///All values inheirited from the sales tagger it came from.
 	var/datum/bank_account/payments_acc = null
-	var/percent_cut = 5
+	var/cut_multiplier = 0.2
```
Ditto here

cacogen committed:
Rounds value in user message

cacogen committed:
Rounds value in similar user message

- Removes improper capitalisation in barcode tag name

cacogen committed:
Defaults to 50% cuts for barcodes, uses initial() if a value isn't input

cacogen committed:
Adds notice spans to examine messages

- Removes ctrl-click message from description and adds it as an variable examine message
- Adds tip about alt-clicking to examine message to change profit split

cacogen committed:
Makes minimum sales cut and maximum sales cut into variables

- Uses these variables to tell the player what range they can set the sales cut in
